lib/Net/Async/Redis.pm

```diff
@@ -1146,8 +1146,8 @@ sub execute_command {
         return $self->stream->write($data)->on_ready($f) if $is_sub_command;
 
         # Void-context write allows IaStream to combine multiple writes on the same connection.
-        push @{$self->{pending}}, [ $cmd, $f ];
         $self->stream->write($data);
+        push @{$self->{pending}}, [ $cmd, $f ];
```

Comment:
If the pending is appended first, and write action failed , for example, redis server stoppped, and then redis server restarted again, then the pending number will be more than the number of redis server replied, because some request missed in the period of redis stopped. So $f will be never resolved.

Moving it after `stream->write` should resolve this problem.